dwl.c

```diff
@@ -1797,6 +1797,7 @@ motionabsolute(struct wl_listener *listener, void *data)
 	struct wlr_pointer_motion_absolute_event *event = data;
 	double lx, ly, dx, dy;
 	wlr_cursor_absolute_to_layout_coords(cursor, &event->pointer->base, event->x, event->y, &lx, &ly);
+	wlr_cursor_warp_closest(cursor, &event->pointer->base, lx, ly);
 	dx = lx - cursor->x;
 	dy = ly - cursor->y;
```

Comment:
Applying this patch may resolve the issue:
```diff
diff --git a/dwl.c b/dwl.c
index 9490b84..f033d8d 100644
--- a/dwl.c
+++ b/dwl.c
@@ -2061,8 +2061,13 @@ motionabsolute(struct wl_listener *listener, void *data)
 	struct wlr_pointer_motion_absolute_event *event = data;
 	double lx, ly, dx, dy;
 	wlr_cursor_absolute_to_layout_coords(cursor, &event->pointer->base, event->x, event->y, &lx, &ly);
-	dx = lx - cursor->x;
-	dy = ly - cursor->y;
+	if (event->pointer->base.type == WLR_INPUT_DEVICE_TOUCH) {
+		dx = lx;
+		dy = ly;
+	} else {
+		dx = lx - cursor->x;
+		dy = ly - cursor->y;
+	}
 
 	motionnotify(event->time_msec, &event->pointer->base, dx, dy, dx, dy);
 }
```

Reply:
What about adding `wlr_cursor_warp_closest(cursor, &event->pointer->base, lx, ly)` in the wrapper function of the touch patch?

Reply:
Thank you for the suggestion.

I modified the "wrapper" function (`touchdown`) to be a copy of the `motionabsolute` function from the pointerconstraints branch.

This works and allows the pointerconstraints branch to remain unchanged.
However, my touchscreen patch is now dependent upon the pointerconstraints patch and that is not a desirable scenario. Each should operate independently of the other unless and until the pointerconstraints patch is merged into mainstream.

Would you consider an addition to the pointerconstraints' dwl.c file like this
`#ifndef POINTERCONSTRAINTS`
`#define POINTERCONSTRAINTS`
`#endif`


(If so, I'll generate the pull request.)

This would allow patches (like my touchscreen) to work with or without pointerconstraints when their own operation is affected by the pointerconstraints patch.

Reply:
> Would you consider an addition to the pointerconstraints' dwl.c file like this
`#ifndef POINTERCONSTRAINTS`
`#define POINTERCONSTRAINTS`
`#endif`
> 

Makes sense

Reply:
That is sufficient to close this issue.
That minor change is made and pushed.
Thanks @sevz17.